From e330444a6fd2ae9ca0065f05cf38798231b9e156 Mon Sep 17 00:00:00 2001
From: Marko Lindqvist <cazfi74@gmail.com>
Date: Mon, 5 Sep 2022 00:13:00 +0300
Subject: [PATCH 53/53] Always free unit build reqs vector in unit_type_free()

It was being skipped for units that do not have
handwritten helptext part.

See osdn #45572

Signed-off-by: Marko Lindqvist <cazfi74@gmail.com>
---
 common/unittype.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/common/unittype.c b/common/unittype.c
index 507d704406..818d85d209 100644
--- a/common/unittype.c
+++ b/common/unittype.c
@@ -2415,9 +2415,10 @@ static void unit_type_free(struct unit_type *punittype)
   if (NULL != punittype->helptext) {
     strvec_destroy(punittype->helptext);
     punittype->helptext = NULL;
-    requirement_vector_free(&(punittype->build_reqs));
   }
 
+  requirement_vector_free(&(punittype->build_reqs));
+
   veteran_system_destroy(punittype->veteran);
   combat_bonus_list_iterate(punittype->bonuses, pbonus) {
     FC_FREE(pbonus);
-- 
2.35.1

